runtime/arc.js

```diff
@@ -159,7 +169,30 @@ class Arc {
     return s;
   }
 
-  connectParticleToView(particleHandle, particle, name, targetView) {
+  instantiate(recipe) {
+    assert(recipe.isResolved(), 'Cannot instantiate an unresolved recipe');
+    let {views, particles, slots} = recipe.mergeInfo(this._activeRecipe);
+    let recipeViewMap = new Map();
+    views.forEach(recipeView => {
+      let view;
+      if (recipeView.create) {
+        view = this.createView(recipeView.type);
+      } else {
+        view = this.viewById(recipeView.id);
+        assert(view, `view '${recipeView.id}' is not registered in arc`);
+      }
+      recipeViewMap.set(recipeView, view);
+    });
+
+    particles.forEach(recipeParticle => this.instantiateParticle(recipeParticle, recipeViewMap));
```

Comment:
just realized that i don't need this map at all, can just call viewById in instantiateParticle.